src/norm/postgres.nim

```diff
@@ -393,8 +393,57 @@ proc update*[T: Model](dbConn; obj: var T) =
 proc update*[T: Model](dbConn; objs: var openArray[T]) =
   ## Update rows for each `Model`_ instance in open array.
 
-  for obj in objs.mitems:
-    dbConn.update(obj)
+  # for obj in objs.mitems:
+  #   dbConn.update(obj)
+
+  # return
+
+  type BulkUpdateValues = Table[string, seq[tuple[id: int64; val: DbValue]]]
+
+  func genBulkUpdateQuery(tableName: string; values: BulkUpdateValues): tuple[qry: string; values: seq[DbValue]] =
+    result.qry = fmt"UPDATE {tableName} SET{'\l'}"
+    var ids: seq[int64]
+    let limit = values.len - 1
+    var
+      i = -1
```

Comment:
This is a weird initial value as the first thing you do with it is `inc` it.

Why not start with 0 and inc it at the end of the loop?